protocol/x/clob/keeper/process_operations_liquidations.go

```diff
@@ -11,6 +11,56 @@ import (
 	satypes "github.com/dydxprotocol/v4-chain/protocol/x/subaccounts/types"
 )
 
+// ValidateLiquidationOrderAgainstProposedMatch performs stateless validation of a liquidation order
+// against a proposed match.
+// An error is returned when
+//   - The CLOB pair IDs of the order and proposed match do not match.
+//   - The perpetual IDs of the order and proposed match do not match.
+//   - The size of the order is less than the size of the proposed match.
+//   - The side of the order and proposed match do not match.
+func (k Keeper) ValidateLiquidationOrderAgainstProposedMatch(
+	ctx sdk.Context,
+	order *types.LiquidationOrder,
+	proposedMatch *types.MatchPerpetualLiquidation,
+) error {
+	if order.GetClobPairId() != types.ClobPairId(proposedMatch.GetClobPairId()) {
+		return sdkerrors.Wrapf(
+			types.ErrClobPairAndPerpetualDoNotMatch,
+			"Order CLOB Pair ID: %v, Match CLOB Pair ID: %v",
+			order.GetClobPairId(),
+			proposedMatch.GetClobPairId(),
+		)
+	}
+
+	if order.MustGetLiquidatedPerpetualId() != proposedMatch.GetPerpetualId() {
+		return sdkerrors.Wrapf(
+			types.ErrClobPairAndPerpetualDoNotMatch,
+			"Order Perpetual ID: %v, Match Perpetual ID: %v",
+			order.MustGetLiquidatedPerpetualId(),
+			proposedMatch.GetPerpetualId(),
+		)
+	}
+
+	if order.GetBaseQuantums() != satypes.BaseQuantums(proposedMatch.TotalSize) {
```

Comment:
I think this is wrong right? this is checking equality instead of ` The size of the order is less than the size of the proposed match.`?

Reply:
Comment is wrong - updated. let me know if that doesn't make sense